### PR TITLE
🌱 Build ironic always from source

### PR DESCRIPTION
**Pull request by mquhuy**
As titled. When integration test is run on top of `main` branch of `ironic-image` repo, ironic image should be built from source.

### CODE REVIEW

**Review by smoshiur1237 — COMMENTED** [review]
/lgtm

File: jenkins/scripts/integration_test.sh
Comment (mboukhalfa) [inline jenkins/scripts/integration_test.sh:57]: Is the REPO_BRANCH defined here

Comment (mquhuy) [reply jenkins/scripts/integration_test.sh:57]: It should be there. It was defined here: https://github.com/metal3-io/project-infra/blob/main/jenkins/jobs/prow_integration_tests.pipeline#L44

File: jenkins/scripts/integration_test.sh
Comment (mboukhalfa) [inline jenkins/scripts/integration_test.sh:58]: Confused which one is used later on `IRONIC_INSTALL_TYPE` or `IRONIC_FROM_SOURCE`

File: jenkins/scripts/integration_test.sh
Comment (tuminoid) [inline jenkins/scripts/integration_test.sh:57]:
```suggestion
if [[ "${REPO_NAME}" == "ironic-image" ]] && [[ "${REPO_BRANCH}" == "main" ]]; then
```

File: jenkins/scripts/integration_test.sh
Comment (mquhuy) [reply jenkins/scripts/integration_test.sh:58]: Setting `IRONIC_INSTALL_TYPE="source"` here will allow it to be handled properly later on, without having to duplicate the logic.

File: jenkins/scripts/integration_test.sh
Comment (mquhuy) [reply jenkins/scripts/integration_test.sh]: Thanks. Done

**Comment by tuminoid** [comment]
/cc @dtantsur @elfosardo 

I want Dmitry and Riccardo to ack this idea as well before merging it. Locking it to source build may its benefits but also downsides we haven't considered.

/hold

**Comment by tuminoid** [comment]
> /cc @dtantsur @elfosardo
> 
> I want Dmitry and Riccardo to ack this idea as well before merging it. Locking it to source build may its benefits but also downsides we haven't considered.
> 
> /hold

OK, discussed in Slack: https://kubernetes.slack.com/archives/CHD49TLE7/p1713856824857309

We should be removing RPM builds all together, only do source, main branch and release as well. It has been a default in Ironic-image for a year now: https://github.com/metal3-io/ironic-image/commit/976769d8730e98bb63411751e21e1d1d8b50d9d9 but of course project-infra is overriding that default right now.

/cc @Rozzii

**Comment by mquhuy** [comment]
> We should be removing RPM builds all together, only do source, main branch and release as well. It has been a default in Ironic-image for a year now: [metal3-io/ironic-image@976769d](https://github.com/metal3-io/ironic-image/commit/976769d8730e98bb63411751e21e1d1d8b50d9d9) but of course project-infra is overriding that default right now.
>
OK. I updated the PR accordingly.

File: jenkins/scripts/integration_test.sh
Comment (tuminoid) [inline jenkins/scripts/integration_test.sh:56]:
```suggestion
if [[ "${REPO_NAME}" == "ironic-image" ]]; then
```
No need to check for the install type, we'll drop the RPM support completely.

Also,
/retitle 🌱 Build ironic always from source

**Review by tuminoid — COMMENTED** [review]
/approve

Looks simple and effective to me.

**Comment by tuminoid** [comment]
> I want Dmitry and Riccardo to ack this idea as well before merging it. Locking it to source build may its benefits but also downsides we haven't considered.
> 
> /hold

Unhold as you see fit.

**Review by dtantsur — APPROVED** [review]

**Pull request merged**
